generators/client-2/templates/src/main/webapp/app/admin/configuration/_configuration.service.ts

```diff
@@ -1,47 +1,49 @@
-(function() {
-    'use strict';
+import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+import 'rxjs/add/operator/toPromise';
+import 'rxjs/add/operator/map';
 
-    angular
-        .module('<%=angularAppName%>.admin')
-        .factory('<%=jhiPrefixCapitalized%>ConfigurationService', <%=jhiPrefixCapitalized%>ConfigurationService);
 
-    <%=jhiPrefixCapitalized%>ConfigurationService.$inject = ['$filter', '$http'];
+@Injectable()
+export class <%=jhiPrefixCapitalized%>ConfigurationService {
 
-    function <%=jhiPrefixCapitalized%>ConfigurationService ($filter, $http) {
-        var service = {
-            get: get,
-            getEnv: getEnv
-        };
-
-        return service;
-
-        function get () {
-            return $http.get('management/configprops').then(getConfigPropsComplete);
+    constructor( private http:Http ){} 
 
+    get () {
+         return this.http.get('management/configprops').toPromise()
+               .then( getConfigPropsComplete );
+               
             function getConfigPropsComplete (response) {
                 var properties = [];
-                angular.forEach(response.data, function (data) {
-                    properties.push(data);
-                });
-                var orderBy = $filter('orderBy');
-                return orderBy(properties, 'prefix');
+                response.data.forEach(prop => {
+                    properties.push(prop);
+                });                
+
+                return properties.sort( (propertyA, propertyB) => {	
```

Comment:
may be we should make orderBy and filter as util services as it would be required often. Can you do a PR later for that?